### ISSUE TITLE
[Bug]: MultimodalVectorStoreIndex loading documents twice when using it with is_image_to_text=True

### ISSUE THREAD

**Issue by jjubric** [run-llama/llama_index#10289]
### Bug Description

I'm using llamaindex to develop a multimodal RAG application. I have two VectorStores, one for text and one for image. I build the index directly from the constructor specifying nodes. These nodes include Text nodes and Image Nodes. When I load the documents it all seems fine, but when looking into Redis the documents are duplicated across the indices, when text documents should only be in text index and image documents only in image index. 

In the code image_nodes have image nodes with text and table_nodes and text_nodes contain both nodes with text. embeddings variable is a AzureOpenAI ada-2 embedding model.

### Version

0.28.post2

### Steps to Reproduce

text_store = RedisVectorStore(index_name=f"text_{index_prefix}", 
                                    index_prefix=index_prefix+"-text", 
                                    redis_url=redis_url
                                    )
    image_store = RedisVectorStore(index_name=f"images_{index_prefix}", 
                                    index_prefix=index_prefix+"-image", 
                                    redis_url=redis_url
                                    )
    
    storage_context = StorageContext.from_defaults(vector_store=text_store, 
                                                   image_store=image_store
                                                )
    service_context = ServiceContext.from_defaults(embed_model=embeddings, 
                                                transformations=[SentenceSplitter.from_defaults()]
                                                )

images = SimpleDirectoryReader(input_dir=images_path).load_data()

image_nodes, table_nodes = summarize_content(images, tables, load_image_descriptions=False)
text_nodes = node_parser.get_nodes_from_documents(texts)
        
multimodal_index = MultiModalVectorStoreIndex(nodes=image_nodes + table_nodes + text_nodes,
                                        storage_context=storage_context,
                                        service_context=service_context,
                                        is_image_to_text=True
                                        )

try:
   storage_context.persist(persist_dir)
except ResponseError:
    pass

**Comment by nerdai** [comment]
Thanks @jjubric, the `image_to_text` boolean variable is used to indicate whether or not the user would like to have "text" representations of the image be the basis of the image embedding. Specifically:

Embedding representations of image:
1. Image -> Image embedding (using CLIP as an example)
2. Image -> Text (i.e., a caption of the image) -> Text embedding 

When `image_to_text = True` as you have here (btw, default for this value is false), then the embedding method 2. is called and in fact will be used to query the images. However, as you pointed out, our code will also store the images in the Image Vector Store (using the text version for its embeddings).

In short, if you don't want this behaviour, you can leave it as default `image_to_text=False`.

**Comment by jjubric** [comment]
Thanks for the help!!
